include/xtensor/xhistogram.hpp

```diff
@@ -16,70 +16,84 @@
 
 #include "xtensor.hpp"
 #include "xsort.hpp"
+#include "xset_operation.hpp"
+#include "xview.hpp"
+
+using namespace xt::placeholders;
 
 namespace xt
 {
+    /**
+     * @ingroup digitize
+     * @brief Return the indices of the bins to which each value in input array belongs.
+     *
+     * @param data The data.
+     * @param bin_edges The bin-edges. It has to be 1-dimensional and monotonic.
+     * @param right Indicating whether the intervals include the right or the left bin edge.
+     * @param assume_sorted Assume data sorted. [default: false]
+     * @return Output array of indices, of same shape as x.
+     */
+    template <class E1, class E2>
+    inline auto digitize(E1&& data, E2&& bin_edges, bool right = false, bool assume_sorted = false)
+    {
+        XTENSOR_ASSERT(bin_edges.dimension() == 1);
+        XTENSOR_ASSERT(bin_edges.size() >= 2);
+        XTENSOR_ASSERT(std::is_sorted(bin_edges.cbegin(), bin_edges.cend()));
+        XTENSOR_ASSERT(xt::amin(data)[0] >= bin_edges[0]);
+        XTENSOR_ASSERT(xt::amax(data)[0] <= bin_edges[bin_edges.size() - 1]);
+
+        if (assume_sorted) {
```

Comment:
- I guess hte condition should be the opposite regarding the line below?
- formatting: the curly brace should be on a new line